src/components/views/login/RegistrationForm.js

```diff
@@ -103,7 +101,7 @@ module.exports = React.createClass({
 
     _doSubmit: function() {
         var promise = this.props.onRegisterClick({
-            username: this.refs.username.value.trim() || this.props.defaultUsername,
+            username: this.refs.username.value.trim() || this.props.guestUsername,
```

Comment:
Please explain how `guestUsername` and `defaultUsername` interact with each other in the `propTypes` for each property. It is not immediately obvious that:
 - The username input box will initially have the `defaultUsername`.
 - Changing the username to a 0-length string will use the `guestUsername`.

A one-liner would make this clear I think.